src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1259,14 +1259,14 @@ async Task StartProcessingInvocationMessages(ChannelReader<InvocationMessage> in
         internal Task RunTimerActions()
         {
             // Don't bother acquiring the connection lock. This is only called from tests.
-            return _state.CurrentConnectionStateUnsynchronized.RunTimerActions();
+            return _state.CurrentConnectionStateUnsynchronized!.RunTimerActions();
         }
 
         // Internal for testing
         internal void OnServerTimeout()
         {
             // Don't bother acquiring the connection lock. This is only called from tests.
-            _state.CurrentConnectionStateUnsynchronized.OnServerTimeout();
+            _state.CurrentConnectionStateUnsynchronized!.OnServerTimeout();
```

Comment:
Could this result in a NullReferenceException? Given that the caller acquires the connection lock after calling OnServerTimeout, it sure looks like it could.

https://github.com/dotnet/aspnetcore/blob/5c0693f9418788e59a5d70dab9536719a84aff00/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L1850-L1857

Any method that references CurrentConnectionStateUnsynchronized without locking because it assumed the caller did so on it's behalf should have `Unsynchronized` in its name and probably call `_state.AssertInConnectionLock()`.

Reply:
Nvm. The comment was right about it being only called from tests. There are multiple OnServerTimeout() methods in the same file.